apollo-runtime/src/main/java/com/apollographql/android/cache/http/CacheInterceptor.java

```diff
@@ -67,33 +51,51 @@ private Response cacheOnlyResponse(Request request) throws IOException {
     return Utils.unsatisfiableCacheRequest(request);
   }
 
-  private Response resolveResponse(Response networkResponse, Response cacheResponse, String cacheKey,
-      HttpCache.CacheControl cacheControl, boolean prefetch) throws IOException {
-    if (networkResponse.isSuccessful()) {
-      cacheResponse.close();
-      if (prefetch) {
-        return prefetch(networkResponse, cacheKey);
-      } else {
-        return cache.cacheProxy(networkResponse, cacheKey)
-            .newBuilder()
-            .cacheResponse(Utils.strip(cacheResponse))
-            .build();
-      }
+  private Response networkFirst(Request request, Chain chain, String cacheKey) throws IOException {
+    Response networkResponse = withServedDateHeader(chain.proceed(request));
+    if (isPrefetchResponse(request)) {
+      return prefetch(networkResponse, cacheKey);
+    } else if (networkResponse.isSuccessful()) {
+      return cache.cacheProxy(networkResponse, cacheKey);
     }
 
-    if (cacheControl == HttpCache.CacheControl.NETWORK_BEFORE_STALE) {
+    Response cacheResponse = cache.read(cacheKey, shouldExpireAfterRead(request));
+    if (cacheResponse == null) {
+      return networkResponse;
+    } else if (!cache.isStale(cacheResponse) || shouldReturnStaleCache(request)) {
       return cacheResponse.newBuilder()
-          .cacheResponse(Utils.strip(cacheResponse))
-          .networkResponse(Utils.strip(networkResponse))
+          .cacheResponse(strip(cacheResponse))
+          .networkResponse(strip(networkResponse))
+          .request(request)
           .build();
     }
 
-    cacheResponse.close();
     return networkResponse.newBuilder()
         .cacheResponse(Utils.strip(cacheResponse))
         .build();
   }
 
+  private Response cacheFirst(Request request, Chain chain, String cacheKey) throws IOException {
+    Response cacheResponse = cache.read(cacheKey, shouldExpireAfterRead(request));
+    if (cacheResponse == null || cache.isStale(cacheResponse)) {
+      Response networkResponse = withServedDateHeader(chain.proceed(request));
```

Comment:
Say you have value object:
class RequestPojo{
String myParam
String timestamp=new Date().getTime();
}

That request will always have a cache miss when trying to figure out if it exists in cache.  A common practice in building caches is to null out all data prior to creating the "Cache Key" in our case it would be prior to the md5 calculation.